### PR TITLE
Yang Set Packing

### LINKED ISSUE #22: Move yang set_packing from old repo to here.
And make it compliant. Should be quit similar to set covering.

### PR DESCRIPTION
**Pull request by mmghannam**
Closes #22

### CODE REVIEW

**Comment by mmghannam** [comment]
This is still missing the simple instance test. I am changing it to a draft PR.

File: geco/mips/set_packing.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    Set Packing, and 0-1 Knapsack Problems", 2020.
+    """
+    return set_packing(m, *yang_parameters(m, seed), name="Yang Set Packing")
+
+
+@py_random_state(-1)
+def yang_parameters(m, seed=0):
+    n = 5 * m
+    costs = [seed.randint(1, 100) for _ in range(n)]
+    nonzero_vars_for_constraint = [
```
Comment (CharJon) [inline geco/mips/set_packing.py]: I think we should name this num_nonzero...

File: geco/mips/set_packing.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    "Learning Generalized Strong Branching for Set Covering,
+    Set Packing, and 0-1 Knapsack Problems", 2020.
+    """
+    return set_packing(m, *yang_parameters(m, seed), name="Yang Set Packing")
+
+
+@py_random_state(-1)
+def yang_parameters(m, seed=0):
+    n = 5 * m
+    costs = [seed.randint(1, 100) for _ in range(n)]
```
Comment (CharJon) [inline geco/mips/set_packing.py:33]: randint(1, 100+1)

File: geco/mips/set_packing.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    nonzero_vars_for_constraint = [
+        seed.randint(2 * n // 25 + 1, 3 * n // 25 - 1) for _ in range(m)
```
Comment (CharJon) [inline geco/mips/set_packing.py:35]: randint(2 * n // 25 + 1, 3 * n // 25 - 1 + 1)

**Review by CharJon — CHANGES_REQUESTED** [review]

**Pull request merged**